Doc/whatsnew/3.11.rst

```diff
@@ -604,6 +604,12 @@ Build Changes
   ``isinf()``, ``isnan()``, ``round()``.
   (Contributed by Victor Stinner in :issue:`45440`.)
 
+* Building Python now requires a C99 ``<math.h>`` header file providing
+  a ``NAN`` constant, or the ``__builtin_nan()`` built-in function. If a
+  platform does not support Not-a-Number (NaN), the ``Py_NO_NAN`` macro can be
+  defined in the ``pyconfig.h`` file.
```

Comment:
It turns out that building under `Py_NO_NAN` doesn't actually work right now. I've opened an issue: https://bugs.python.org/issue46656